crates/ra-salsa/Cargo.toml

```diff
@@ -1,5 +1,5 @@
 [package]
-name = "salsa"
+name = "ra-salsa"
```

Comment:
Can we keep this as salsa? (that way we dont create a new autopublished crate for the new name), given the `[lib].name` section this should be fine I think? If not then thats fine, its just a minor nit (same for the macros crate)

Reply:
oh, good point! i added a `package` specifier wherever Salsa is used, so that'll avoid creating new autopublished crates.

Reply:
A small confusion caused by this 😅 #18297

Reply:
whoops!